javaparser-core/src/main/java/com/github/javaparser/ast/body/ClassOrInterfaceDeclaration.java

```diff
@@ -148,6 +156,20 @@ public ClassOrInterfaceDeclaration setImplementedTypes(final NodeList<ClassOrInt
         return this;
     }
 
+    @Generated("com.github.javaparser.generator.core.node.PropertyGenerator")
+    public ClassOrInterfaceDeclaration setPermittedTypes(final NodeList<ClassOrInterfaceType> permittedTypes) {
```

Comment:
Can you confirm for me that the generated properties are actually generated?

Reply:
I did run the metamodel and core generator scripts, which added the getters/setters after I added the field. But that didn't modify the `@AllFieldsConstructor`-constructor or the main-constructor. Those I had to update manually.

Also the generator updated pretty much all files, mostly by adding/removing whitespace in imports and such. So I only kept the changes related to the sealed/non-sealed logic and reverted everything else.

Reply:
I admit that I don't know this part very well (generator) but what seems curious to me, if I understand your answer correctly, is that the generator generates the following code but does not generate the property "permittedTypes".

    this.permittedTypes.setParentNode(null);

For my information, how did you know you had to run the generators and manipulate the metamodel? Did you consult any documentation?

Reply:
I didn't find any documentation for the generator (only IDE-setup, coding-guidelines, etc.). After adding the `permittedTypes` field I just looked around how to invoke the generator for the accessors and other functions. This lead me to the `javaparser-core-generators` module, the `run-generators` Maven-profile and finally the `run_core_generators.sh` script, which contains the comment
```
# Runs all the code generators.
# If the node structure was changed, run_metamodel_generator.sh first!
```
This is essentially the "documentation" to first run the metamodel-generator and then the code-generator. (Plus a bit of inductive reasoning, as I've seen similar projects and code-generators before.)

So, yes, after adding the `permittedTypes` field, running the generators generated the matching accessors and other utility-functions in `ClassOrInterfaceDeclaration`, updated various visitor- and observer-classes, and the metamodel to each include the new field.
I'm not sure, why the `MainConstructorGenerator` didn't update the "main constructor" properly, as I had to update this one manually.

---

Also the generators changed the formatting of all files, that contained generated code by stripping out empty lines in the import statements, so from:
```java
import com.github.javaparser.resolution.Resolvable;
import com.github.javaparser.resolution.declarations.ResolvedReferenceTypeDeclaration;

import java.util.Optional;
import java.util.function.Consumer;

import static com.github.javaparser.utils.Utils.assertNotNull;
```
to
```java
import com.github.javaparser.resolution.Resolvable;
import com.github.javaparser.resolution.declarations.ResolvedReferenceTypeDeclaration;
import java.util.Optional;
import java.util.function.Consumer;
import static com.github.javaparser.utils.Utils.assertNotNull;
```

I reverted these changes, as I didn't want to inflate the PR with hundreds of reformatted, but unrelated, java files.

Reply:
The only documentation I found on this topic is here. Your suggestions for improving it are welcome. https://javaparser.org/code-generation-and-maven-in-javaparser/

Reply:
I did look on the javaparser website, specifically https://javaparser.org/contributing.html , but this only links back to various github pages.
Honestly, without knowing that the code-generator exists I wouldn't have looked outside the contributing-section.

The contributing-section links to https://github.com/javaparser/javaparser/blob/master/CONTRIBUTING.md, which has links to the github wiki. IMO, the wiki is the correct place to document how to use javaparser and how to contribute. Or at least there should be pages in the wiki that link back to specific javaparser.org blog-articles, so that users/contributers can find the information, without already knowing where to look. 

The blog just isn't a good place for technical documentation, but I also don't know the "official" status of javaparser.org. To me a lot of information and documentation seems to be spread across the javaparser.org and tomassetti.me website/blog and the github wiki and documents, which makes it hard to find specific documentation.